cleverhans/attacks_tf.py

```diff
@@ -1842,7 +1872,12 @@ def cond(i, *_):
     # project_perturbation
     check_diff = tf.no_op()
 
-  with tf.control_dependencies([check_diff]):
+  if clip_min is None or clip_max is None:
+    raise NotImplementedError("SPSA only supports clipping for now")
```

Comment:
this code is inside pgd_attack function.
Why you referring to SPSA attack in the text of the error?

Reply:
The `pgd_attack` has a confusing name.

You're proving the point I made in this issue :)  https://github.com/tensorflow/cleverhans/issues/780